google/cloud/storage/client.py

```diff
@@ -530,7 +530,7 @@ def download_blob_to_file(self, blob_or_uri, file_obj, start=None, end=None):
         try:
             blob_or_uri.download_to_file(file_obj, client=self, start=start, end=end)
         except AttributeError:
-            blob = Blob.from_string(blob_or_uri)
+            blob = Blob.from_string(blob_or_uri, client=self)
```

Comment:
was this an accidental miss in implementation? Seems like it but want to confirm.

Reply:
Looking at it again, client is passed below to download_to_file(). The change doesn't look necessary.

Reply:
Yes, but the problem arise here , if i didn't pass the client so bucket class has received `None`
https://github.com/googleapis/python-storage/blob/18b1e07d66b6dfde2de5f1d97a09ab0f25db1d1b/google/cloud/storage/blob.py#L351

 so when call  `_require_client()` method it find the client object which  passed in` download_to_file()` so `hostname=self.client._connection.API_BASE_URL` line hit the error as self.client found `None`

Reply:
Gotcha, when I raised that you should revert my last comment, I was referring to updating the method to accept the client because even though `_get_transport` calls `_require_client()` it doesn't update `self.client`.

I think you may want to pass client returned from `_require_client()` to `_get_download_url()`.

Thank you for your patience.